src/ucp/rndv/rndv.h

```diff
@@ -75,4 +90,14 @@ void ucp_rndv_receive(ucp_worker_h worker, ucp_request_t *rreq,
 ucs_status_t ucp_rndv_rts_handle_status_from_pending(ucp_request_t *sreq,
                                                      ucs_status_t status);
 
+static UCS_F_ALWAYS_INLINE int
+ucp_rndv_rts_is_tag(const ucp_rndv_rts_hdr_t *rts_hdr)
+{
+    return rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG ||
+           /* Wire compatibility with old versions of RNDV protocol which uses
+            * status instead of flags, only TAG operations could be used then */
+           ucs_unlikely((rts_hdr->status == UCS_OK) ||
```

Comment:
maybe unlikely only for `UCS_ERR_CANCELED`?

Reply:
new protocols should not have `status` included, only flags

Reply:
does it mean that it uses different header?

Reply:
After this PR the headers are almost the same, expect the last byte (in old version it is `ucs_status_t`, in current version it is `uint8_t` flags)

Reply:
I see, maybe add `UCP_RNDV_RTS_FLAG_TAG_COMPAT_VALUE_OK = 0` and `UCP_RNDV_RTS_FLAG_TAG_COMPAT_VALUE_CANCELED = UCS_ERR_CANCELED` to flags?
also, in theory, some exotic compiler can produce padding in union of `uint8_t` and packed enum `ucs_status_t`, so IMO better to completely avoid ucs_status_t in new header

Reply:
Yes, we can avoid `ucs_status_t` to be optimized in case of "exotic" compilers.

Reply:
Yes, I agree, this also can lead to wire compatibility problems

Reply:
done